tools/_setup_generation/step_gallery.py

```diff
@@ -0,0 +1,119 @@
+# ##########################################################################################
+# Step to generate the list of gallery items in the gallery index page
+#
+# The gallery index page is generated from all the gallery.
+# Each demo is an index.md file with a header containing the necessary information to
+# generate the list of items in the gallery index page.
+#
+# The header of an item is a multiline header starting with '---' and ending with '---'.
+# The header contains the following information:
+# - title: The title of the item
+# - category: The category of the item (tutorials, gallery or tips)
+# - type: The type of the item (code, video or article)
+# - data-keywords: A comma separated list of keywords
+# - short_description: A short description of the item
+#
+# ##########################################################################################
+import os
+import re
+from typing import List
+
+from .setup import SetupStep, Setup
+from .items import FolderItem, FileItem, Item
+from .items.exceptions import WrongHeader, NoHeader, NoIndexFile
+
+
+class GalleryStep(SetupStep):
+
+    # The name of the gallery folder
+    GALLERY_FOLDER_NAME = "gallery"
+
+    # The placeholders in the gallery index file to be replaced by the generated content
+    GALLERY_PLACEHOLDER = r"\[LIST_FOR_GALLERY\]"
+
+    def __init__(self):
+        self.GALLERY_FOLDER_PATHS = None
+        self.GALLERY_INDEX_PATH = None
+        self.GALLERY_INDEX_TPL_PATH = None
+        self.gallery = ""
+
+
+    def enter(self, setup: Setup):
+        self.GALLERY_FOLDER_PATH = os.path.join(setup.docs_dir, self.GALLERY_FOLDER_NAME)
+        self.GALLERY_INDEX_PATH = os.path.join(setup.docs_dir, self.GALLERY_FOLDER_NAME, "index.md")
+        self.GALLERY_INDEX_TEMPLATE_PATH = os.path.join(setup.docs_dir, self.GALLERY_FOLDER_NAME, "index.md_template")
```

Comment:
```suggestion
        self.GALLERY_INDEX_TEMPLATE_PATH = os.path.join(self.GALLERY_FOLDER_PATH, "index.md_template")
```

Reply:
Creating a variable for the template source should not be a good idea.
You should, in the code, depend on the actual output file name (self.GALLERY_INDEX_PATH here) and add '_template' when you need it. So both file names are tightly coupled with no risk of diverging.

Reply:
I think I have corrected it now